src/v/datalake/translation/deps.h

```diff
@@ -60,12 +91,14 @@ class translator_mem_tracker : public writer_mem_tracker {
 public:
     explicit translator_mem_tracker(
       scheduling::reservations_tracker& scheduling_reservations)
-      : _reservations_tracker(scheduling_reservations) {}
+      : _reservations_tracker(scheduling_reservations)
+      , _disk(scheduling_reservations) {}
 
     ss::future<reservation_error>
     reserve_bytes(size_t, ss::abort_source&) noexcept override;
     ss::future<> free_bytes(size_t, ss::abort_source&) override;
     void release() override;
+    writer_disk_tracker& disk() override;
```

Comment:
a bit confused why disk tracker is accessed via (and is part of) mem tracker ❓

Reply:
does the commit message answer your question? https://github.com/redpanda-data/redpanda/pull/25766/commits/ad624852f9ec222c345d56f81173f6fa0b6ac465